### PR TITLE
Cyborgs get radio for their module

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Cyborgs get radio keys for their job (ie miners get Supply, Science, and Explo (if they want to help explocrew)

## Why It's Good For The Game
Makes it far easier as a borgo to support your department, know where people are, etc

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
image
image

</details>

## Changelog
:cl:
add: borgs get relevant radio keys
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 1]
```diff
@@ -290,6 +292,8 @@
 	moduleselect_icon = "standard"
 	hat_offset = -3
 
+	radio_channels = (RADIO_CHANNEL_AI_PRIVATE)
```
Comment (Sarchutar) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]:
```suggestion
	radio_channels = list(RADIO_CHANNEL_AI_PRIVATE)
```

**Comment by Rukofamicom** [comment]
Screenshot_20220911-204504

These are the channels borgs should be getting per #7458.

Civilian channel doesn't exist yet, but those specific borgs don't have any department affiliation and don't need any extra channels

**Comment by etherware-novice** [comment]
but mining keys grant science..

File: code/game/objects/items/devices/radio/radio.dm [hunk 2]
```diff
@@ -409,6 +409,14 @@
 /obj/item/radio/borg/Initialize(mapload)
 	. = ..()
 
+/obj/item/radio/borg/recalculateChannels()
+	var/mob/living/silicon/robot/R = loc
+	if(istype(loc))
+		for(var/channel in R.module.radio_channels)
+			channels[channel] = 1
```
Comment (itsmeow) [inline code/game/objects/items/devices/radio/radio.dm:416]: this should be TRUE, not 1, right?

File: code/game/objects/items/devices/radio/radio.dm [hunk 3]
```diff
@@ -409,6 +409,14 @@
 /obj/item/radio/borg/Initialize(mapload)
 	. = ..()
 
+/obj/item/radio/borg/recalculateChannels()
+	var/mob/living/silicon/robot/R = loc
+	if(istype(loc))
```
Comment (itsmeow) [inline code/game/objects/items/devices/radio/radio.dm:414]: `istype(R)`

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 4]
```diff
@@ -448,6 +459,8 @@
 	can_be_pushed = FALSE
 	hat_offset = -2
 
+	radio_channels = list(RADIO_CHANNEL_SERVICE)  // temporary until civilian channel
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:462]: just remove this

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 5]
```diff
@@ -711,6 +736,8 @@
 	can_be_pushed = FALSE
 	hat_offset = 3
 
+	radio_channels = list(RADIO_CHANNEL_SYNDICATE, RADIO_CHANNEL_MEDICAL)
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:739]: remove med

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 6]
```diff
@@ -745,6 +772,8 @@
 	hat_offset = -4
 	canDispose = TRUE
 
+	radio_channels = list(RADIO_CHANNEL_SYNDICATE, RADIO_CHANNEL_ENGINEERING)
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:775]: remove eng

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 7]
```diff
@@ -600,6 +619,8 @@
 	cyborg_base_icon = "borgi"
 	moduleselect_icon = "standard"
 
+	radio_channels = list(RADIO_CHANNEL_COMMAND)  // to be kidnapped by the HoP
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:622]: remove

File: code/game/objects/items/devices/radio/radio.dm [hunk 8]
```diff
@@ -409,6 +409,14 @@
...
+		for(var/channel in R.module.radio_channels)
+			channels[channel] = 1
+	. = ..()
```
Comment (itsmeow) [inline code/game/objects/items/devices/radio/radio.dm:417]: this doesn't clear channels, so if you change modules back you don't lose any channel

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/radio/radio.dm (hunk 2 above)
Comment (etherware-novice) [reply code/game/objects/items/devices/radio/radio.dm:416]: idk the original radio code used 1

File: code/game/objects/items/devices/radio/radio.dm (hunk 8 above)
Comment (etherware-novice) [reply code/game/objects/items/devices/radio/radio.dm:417]: well how do i fricking do that

**Pull request closed without merging**